components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3566,12 +3560,7 @@ where
             // [entries...][the entry at `compact_idx`][the last entry][new compaction entry]
             //             |-------------------- entries will be left ----------------------|
             return;
-        } else if replicated_idx - first_idx < self.ctx.cfg.raft_log_gc_threshold
-            && self.fsm.skip_gc_raft_log_ticks < self.ctx.cfg.raft_log_reserve_max_ticks
-        {
-            // Logs will only be kept `max_ticks` * `raft_log_gc_tick_interval`.
-            self.fsm.skip_gc_raft_log_ticks += 1;
-            self.register_raft_gc_log_tick();
+        } else if replicated_idx - first_idx < self.ctx.cfg.raft_log_gc_threshold {
```

Comment:
This code is unnecessary. If it's true, the previous condition `replicated_idx < first_idx` must be true.

Reply:
There are tests rely on `threshold` here to prevent from raft log compaction. So I kept it and just set it to 1 as default.

Reply:
Oh, I misread it.

Reply:
Hmmm, maybe it is wasteful to propose a compact log when the log count is too small.

Reply:
With RaftEngine, old log entries will consume IO during rewrite. So, with RaftEngine, it's good that compact old log entries ASAP.

But in any circumstance, raftstore doesn't need to skip ticks. `raft-log-gc-threshold` is enough.

Reply:
Can we cleanup the `raft_log_gc_threshold` config and update the test using failpoint to achieve the same?

Reply:
That's a good idea. Let me give it a try.

Reply:
I lean to tweaking it's default value rather than removing it completely. Also for raft engine, rewrite happens fairly rare, I wouldn't be so sure about setting it to 1.